frontend/src/container/RoutingPolicies/useRoutingPolicies.ts

```diff
@@ -0,0 +1,238 @@
+	const [isDeleteModalOpen, setIsDeleteModalOpen] = useState(false);
+	const [
+		policyDetailsModalState,
+		setPolicyDetailsModalState,
+	] = useState<PolicyDetailsModalState>({
+		mode: null,
+		isOpen: false,
+	});
+	const [
+		selectedRoutingPolicy,
+		setSelectedRoutingPolicy,
+	] = useState<RoutingPolicy | null>(null);
+
+	// Routing Policies list
+	const {
+		data: routingPolicies,
+		isLoading: isLoadingRoutingPolicies,
+		isError: isErrorRoutingPolicies,
+	} = useGetRoutingPolicies();
+
+	const routingPoliciesData = useMemo(() => {
+		const unfilteredRoutingPolicies = mapApiResponseToRoutingPolicies(
+			routingPolicies,
+		);
+		return unfilteredRoutingPolicies.filter((routingPolicy) =>
+			routingPolicy.name.toLowerCase().includes(searchTerm.toLowerCase()),
+		);
+	}, [routingPolicies, searchTerm]);
+
+	// Channels list
+	const {
+		data,
+		isLoading: isLoadingChannels,
+		isError: isErrorChannels,
+		refetch: refetchChannels,
+	} = useQuery<SuccessResponseV2<Channels[]>, APIError>(['getChannels'], {
+		queryFn: () => getAllChannels(),
+	});
+	const channels = data?.data || [];
+
+	const refreshChannels = (): void => {
+		refetchChannels();
+	};
+
+	// Handlers
+	const handlePolicyDetailsModalOpen = (
+		mode: PolicyDetailsModalMode,
+		routingPolicy: RoutingPolicy | null,
+	): void => {
+		if (routingPolicy) {
+			setSelectedRoutingPolicy(routingPolicy);
+		}
+		setPolicyDetailsModalState((prev) => ({
+			...prev,
+			isOpen: true,
+			mode,
+		}));
+	};
+
+	const handlePolicyDetailsModalClose = (): void => {
+		setSelectedRoutingPolicy(null);
+		setPolicyDetailsModalState((prev) => ({
+			...prev,
+			isOpen: false,
+			mode: null,
+		}));
+	};
+
+	const handleDeleteModalOpen = (routingPolicy: RoutingPolicy): void => {
+		setSelectedRoutingPolicy(routingPolicy);
+		setIsDeleteModalOpen(true);
+	};
+
+	const handleDeleteModalClose = (): void => {
+		setSelectedRoutingPolicy(null);
+		setIsDeleteModalOpen(false);
+	};
+
+	// Create Routing Policy
+	const {
+		mutate: createRoutingPolicy,
+		isLoading: isCreating,
+	} = useCreateRoutingPolicy();
+
+	// Update Routing Policy
+	const {
+		mutate: updateRoutingPolicy,
+		isLoading: isUpdating,
+	} = useUpdateRoutingPolicy();
+
+	// Policy Details Modal Action (Create or Update)
+	const handlePolicyDetailsModalAction = (
+		mode: PolicyDetailsModalMode,
+		routingPolicyData: {
+			name: string;
+			expression: string;
+			channels: string[];
+		},
+	): void => {
+		if (mode === 'create') {
+			createRoutingPolicy(
+				{
+					payload: mapRoutingPolicyToCreateApiPayload(
+						routingPolicyData.name,
+						routingPolicyData.expression,
+						routingPolicyData.channels,
+					),
+				},
+				{
+					onSuccess: () => {
+						toast.success('Routing policy created successfully');
+						queryClient.invalidateQueries(REACT_QUERY_KEY.GET_ROUTING_POLICIES);
+						handlePolicyDetailsModalClose();
+					},
+					onError: () => {
+						toast.error('Failed to create routing policy');
```

Comment:
Can we show the actual error response from API here? Example, try creating a policy without name, it should throw bad request error, in which case we should show the error from backend so that it is actionable for user.

Reply:
Yes should be doable